nnfabrik/main.py

```diff
@@ -351,3 +351,99 @@ class Seed(dj.Manual):
     definition = """
     seed:   int     # Random seed that is passed to the model- and dataset-builder
     """
+
+
+def custom_nnfabrik(
+    schema,
+    use_common_fabrikant=True,
+    use_common_seed=False,
+    module_name=None,
+    context=None,
+    spawn_existing_tables=False,
```

Comment:
I'm not sure if we ever agreed on this as a common practice, but I think type-hints would be very useful here (especially for `context` and `schema`).

Reply:
Yeah I still don't feel too sure about using type hints all the time. But if we are going to do it, then I want for us to do it properly and stick to the full convention of typing. I'll go ahead and add type hints to the best completeness I can now.